megamek/src/megamek/client/bot/princess/PathRanker.java

```diff
@@ -250,8 +249,8 @@ private List<MovePath> validatePaths(List<MovePath> startingPathList, Game game,
      * @return "Best" out of those paths
      */
     @Override
-    public @Nullable RankedPath getBestPath(List<RankedPath> ps) {
-        return ps.isEmpty() ? null : Collections.max(ps);
+    public @Nullable RankedPath getBestPath(TreeSet<RankedPath> ps) {
+        return ps.isEmpty() ? null : ps.first();
```

Comment:
TreeSet is ordered, this is implemented with reversed ordering so the first element is always the best ranked movepath